sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java

```diff
@@ -66,7 +66,7 @@ private IndexMetaData purgeDynamicStringTemplate(IndexMetaData indexMetaData) {
     @VisibleForTesting
     MappingMetaData purgeDynamicStringTemplate(MappingMetaData mappingMetaData, String indexName) {
         Map<String, Object> oldMapping = mappingMetaData.getSourceAsMap();
-        HashMap<String, Object> newMapping = new HashMap<>(oldMapping.size());
+        Map<String, Object> newMapping = new LinkedHashMap<>(oldMapping.size());
```

Comment:
minor nitpick: could be defined of `HashMap` as we know its type (no need to just expect an interface)

Reply:
If this won't be backported, you could also use `var` :-)

Reply:
I thought its best practice in java to program against interfaces.

Reply:
If the var is exposed and only the behaviour of an interface is required, sure.

Reply:
How about "var newMapping" ? :-)